### PR TITLE
Remove Restore tab from Manage Brave Rewards dialog

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/26338

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A
  
## UI:

Screenshot 2022-11-01 105717

## Test Plan:

- Clean profile
- Enable Rewards
- Visit brave://rewards
- Click "Manage Brave Rewards" button
- Verify that Manage Brave Rewards dialog only has two tabs: Backup and Reset

### CODE REVIEW

**Comment by szilardszaloki** [comment]
@emerick Left comments in `#rewards-dev`.

File: components/brave_rewards/resources/page/components/pageWallet.tsx [hunk 1]
```diff
@@ -86,13 +83,6 @@ class PageWallet extends React.Component<Props, State> {
     })
   }
 
-  onModalBackupOnRestore = (key: string | MouseEvent) => {
-    if (typeof key === 'string' && key.length > 0) {
-      key = this.pullRecoveryKeyFromFile(key)
```
Comment (szilardszaloki) [inline components/brave_rewards/resources/page/components/pageWallet.tsx:91]: I think we can now remove `pullRecoveryKeyFromFile`, too.

File: components/brave_rewards/resources/page/components/pageWallet.tsx [hunk 2]
```diff
@@ -86,13 +83,6 @@ class PageWallet extends React.Component<Props, State> {
...
-      this.actions.recoverWallet(key)
```
Comment (szilardszaloki) [inline components/brave_rewards/resources/page/components/pageWallet.tsx:92]: I think we can also remove `recoverWalletData()` in `brave_rewards_page.tsx`, `onRecoverWalletData()` in `rewards_actions.ts`, even `types.ON_RECOVER_WALLET_DATA` and the corresponding code in `wallet_reducer.ts`.

File: components/brave_rewards/resources/page/components/pageWallet.tsx [hunk 3]
```diff
@@ -166,21 +156,6 @@ class PageWallet extends React.Component<Props, State> {
     })
   }
 
-  toggleQRCodeModal = () => {
-    // If we are opening the QR code panel, then request a payment ID if we do
-    // not already have one and close the backup/restore modal.
-    if (!this.state.modalQRCode) {
-      if (!this.props.rewardsData.paymentId) {
-        this.actions.getPaymentId()
```
Comment (szilardszaloki) [inline components/brave_rewards/resources/page/components/pageWallet.tsx:174]: It seems that `getPaymentId()` in `rewards_actions.ts` is no longer used, also `types.GET_PAYMENT_ID` and the corresponding handler code in `rewards_reducer.ts`.

**Comment by szilardszaloki** [comment]
@emerick It seems we can remove the entire `RecoverWallet` (`brave_rewards.recoverWallet`) flow from the backend (from `RewardsDOMHandler`, `RewardsServiceImpl`, etc., and from the ledger completely).

**Review by zenparsing — APPROVED** [review]
Front-end changes look good.

File: components/brave_rewards/resources/page/components/pageWallet.tsx (hunk 1 above)
Comment (emerick) [reply components/brave_rewards/resources/page/components/pageWallet.tsx:91]: Thanks, removed!

File: components/brave_rewards/resources/page/components/pageWallet.tsx (hunk 2 above)
Comment (emerick) [reply components/brave_rewards/resources/page/components/pageWallet.tsx:92]: Removed.

File: components/brave_rewards/resources/page/components/pageWallet.tsx (hunk 3 above)
Comment (emerick) [reply components/brave_rewards/resources/page/components/pageWallet.tsx:174]: Removed.

File: ios/browser/api/ledger/brave_ledger.h [hunk 4]
```diff
@@ -123,12 +115,6 @@ OBJC_EXPORT
 - (void)pendingContributionsTotal:(void (^)(double amount))completion
     NS_SWIFT_NAME(pendingContributionsTotal(completion:));
 
-/// Links a desktop brave wallet given some payment ID
-- (void)linkBraveWalletToPaymentId:(NSString*)paymentId
-                        completion:(void (^)(LedgerResult result,
-                                             NSString* drainID))completion
-    NS_SWIFT_NAME(linkBraveWallet(paymentId:completion:));
```
Comment (szilardszaloki) [inline ios/browser/api/ledger/brave_ledger.h:130]: I'm wondering if [this](https://github.com/brave/brave-ios/blob/65f165a370cdaf56ec8d383272f8f7b8dca17feb/Client/Frontend/Brave%20Rewards/Transfer/WalletTransferViewController.swift#L46) has to do something with this line.

**Review by szilardszaloki — APPROVED** [review]
Another ~2k LOC `==>` 🗑️🎉

File: ios/browser/api/ledger/brave_ledger.h (hunk 4 above)
Comment (kylehickinson) [reply ios/browser/api/ledger/brave_ledger.h:130]: Yeah thats just old UI for linking the brave wallet that is no longer shown, we will clean this up on iOS side when this change reaches brave-ios

**Review by kylehickinson — APPROVED** [review]
iOS++

**Pull request merged**